### PR TITLE
Sm/yarn dedupe

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
1. --patch option for latest rc builds
2. yarn-deduplicate to reduce dependency size and help path lengths 
3. bump dev-scripts
### What issues does this PR fix or reference?
@W-10220681@

### CODE REVIEW

**Review by WillieRuemmele — APPROVED** [review]

File: src/commands/cli/latestrc/build.ts [hunk 1]
```diff
@@ -45,7 +48,7 @@ export default class build extends SfdxCommand {
     // get the current version and implement the patch version for a default rc build
     const repo = await SinglePackageRepo.create({ ux: this.ux });
 
-    const nextRCVersion = repo.package.getNextRCVersion(this.flags.rctag);
+    const nextRCVersion = repo.package.getNextRCVersion(this.flags.rctag, this.flags.path);
```
Comment (RodEsp) [inline src/commands/cli/latestrc/build.ts:51]:
```suggestion
    const nextRCVersion = repo.package.getNextRCVersion(this.flags.rctag, this.flags.patch);
```

File: package.json [hunk 2]
```diff
@@ -32,7 +32,8 @@
     "shelljs": "^0.8.4",
     "standard-version": "^9.0.0",
     "strip-ansi": "6.0.1",
-    "tslib": "^2"
+    "tslib": "^2",
+    "yarn-deduplicate": "^3.1.0"
```
Comment (RodEsp) [inline package.json:36]: You can also suppress the `npx` install prompt with the `-y` or `--yes` flag.

**Review by RodEsp — APPROVED** [review]

**Comment by mshanemc** [comment]
QA: I used it for this: https://github.com/salesforcecli/sfdx-cli/pull/448 (not a very good test)

**Comment by mshanemc** [comment]
QA:
ran `../plugin-release-management/bin/run  cli:latestrc:build --patch`
https://github.com/salesforcecli/sfdx-cli/pull/464/files

**Comment by RodEsp** [comment]
Ship it :shipit:!

**Pull request merged**